BootloaderCorePkg/Library/PciEnumerationLib/PciIov.c

```diff
@@ -68,6 +70,16 @@ PciIovParseVfBar (
     return Offset + 4;
   }
 
+  //
+  // All child devices under PPB must be in scope of PPB's decode space.
+  //
+  Parent = PciIoDevice->Parent;
+  if ((Parent != NULL) && (IS_PCI_BRIDGE (&Parent->Pci))) {
+    DecodeMask = Parent->Decodes;

```

Comment:
Do you think it might be easier to just use DecodeMask for the control? 
EnumPolicy can be used to initialize the top level root bridge DecodeMask.
And then everyone needs to AND with its parent all the way to the root bridge to get its decodes

Reply:
Consider  Bridge1 (Decoding PMEM64/PMEM32) behind Bridge2 (Decoding PMEM32 only),  the device decode should be AND for all its parents, not just its direct parent.

Reply:
> Do you think it might be easier to just use DecodeMask for the control?
> EnumPolicy can be used to initialize the top level root bridge DecodeMask.
> And then everyone needs to AND with its parent all the way to the root bridge to get its decodes

That's good idea! Let me use EnumPolicy for root bridge DecodeMask and simplify downgrade control.

Reply:
> Consider Bridge1 (Decoding PMEM64/PMEM32) behind Bridge2 (Decoding PMEM32 only), the device decode should be AND for all its parents, not just its direct parent.

You are right. Let me fix this. Thanks!